src/Compilers/CSharp/Portable/Lowering/ClosureConversion/ExpressionLambdaRewriter.cs

```diff
@@ -1054,58 +1140,71 @@ private BoundExpression VisitUnaryOperator(BoundUnaryOperator node)
                     break;
                 default:
                     throw ExceptionUtilities.UnexpectedValue(op);
             }
 
+            if ((object)node.MethodOpt == null)
+            {
+                switch (opFactory)
+                {
+                    case WellKnownMember.System_Linq_Expressions_Expression__NegateChecked_Expression_MethodInfo:
+                        opFactory = WellKnownMember.System_Linq_Expressions_Expression__NegateChecked_Expression;
+                        break;
+
+                    case WellKnownMember.System_Linq_Expressions_Expression__Negate_Expression_MethodInfo:
+                        opFactory = WellKnownMember.System_Linq_Expressions_Expression__Negate_Expression;
+                        break;
+
+                    case WellKnownMember.System_Linq_Expressions_Expression__Not_Expression_MethodInfo:
+                        opFactory = WellKnownMember.System_Linq_Expressions_Expression__Not_Expression;
+                        break;
+
+                    default:
+                        throw ExceptionUtilities.UnexpectedValue(opFactory);
+                }
+            }
+
             if (node.OperatorKind.OperandTypes() == UnaryOperatorKind.Enum && (opKind & UnaryOperatorKind.Lifted) != 0)
             {
                 Debug.Assert((object)node.MethodOpt == null);
                 var promotedType = PromotedType(arg.Type.StrippedType().GetEnumUnderlyingType());
                 promotedType = _nullableType.Construct(promotedType);
                 loweredArg = Convert(loweredArg, arg.Type, promotedType, isChecked, false);
-                var result = ExprFactory(opname, loweredArg);
+                var result = _bound.StaticCall(opFactory, loweredArg);
                 return Demote(result, node.Type, isChecked);
             }
 
             return ((object)node.MethodOpt == null)
-                ? ExprFactory(opname, loweredArg)
-                : ExprFactory(opname, loweredArg, _bound.MethodInfo(node.MethodOpt, _bound.WellKnownType(WellKnownType.System_Reflection_MethodInfo)));
+                ? _bound.StaticCall(opFactory, loweredArg)
+                : _bound.StaticCall(opFactory, loweredArg, _bound.MethodInfo(node.MethodOpt, _bound.WellKnownType(WellKnownType.System_Reflection_MethodInfo)));
         }
 
         // ======================================================
 
-        private BoundExpression ExprFactory(string name, params BoundExpression[] arguments)
-        {
-            return _bound.StaticCall(ExpressionType, name, disallowExpandedNonArrayParams: true, arguments);
-        }
-
-        private BoundExpression ExprFactory(string name, ImmutableArray<TypeSymbol> typeArgs, params BoundExpression[] arguments)
-        {
-            return _bound.StaticCall(_ignoreAccessibility ? BinderFlags.IgnoreAccessibility : BinderFlags.None, ExpressionType, name, disallowExpandedNonArrayParams: true, typeArgs, arguments);
```

Comment:
> It looks like we no longer use `_ignoreAccessibility`. Is that an unintended behavior change?

Definitions of well known types and members must be accessible. I am not sure why this field was used. I assume that, if there were specific scenarios in mind, there would be tests that would fail. 

> Also consider removing the unused field.

Will do.